File: CSE 241 - Object Oriented Programming/Homework 5/Homework/BoardArray2D.cpp
#include "BoardArray2D.h"

BoardArray2D::~BoardArray2D(){
	for(int i = 0; i < boardSizey; ++i) delete[] board[i];
	delete[] board;
}

void BoardArray2D::setCellValue(int y, int x, int value){
	board[y][x] = value;
}

void BoardArray2D::setSize(int sizeX, int sizeY){
	for(int i = 0; i < boardSizey; ++i) delete[] board[i];
	delete[] board;
	
	boardSizex = sizeX;
	boardSizey = sizeY;
	
	board = new int *[sizeY];
	for(int i = 0; i < sizeY; ++i) board[i] = new int[sizeX];
}

int BoardArray2D::operator () (int y, int x){
	if(y < boardSizey && x < boardSizex){
		return board[y][x];
  } else{
		exit(0);
  }
}

File: CSE 241 - Object Oriented Programming/Homework 7/Homework/BoardArray2D.java
/**
 * A board class that represents the board as 2D array
 * @author Berke Belgin
 */
public class BoardArray2D extends AbstractBoard{
	private int[][] board;
	
	public BoardArray2D(){
		super();
	}
	public BoardArray2D(int y, int x){
		super(y, x);
	}
	
	protected void finalize(){
		super.finalize();
	}
	
	public void setCell(int y, int x, int value){
		board[y][x] = value;
	}
	public void setSize(int sizeY, int sizeX){
		board = new int[sizeY][sizeX];
		
		this.sizeY = sizeY;
		this.sizeX = sizeX;
	}
	public int cell(int y, int x){
		if(y < sizeY && x < sizeX){
			return board[y][x];
		} else{
			System.exit(0);
			return 0;
		}
	}
}
